test/distributed/_composable/test_fully_shard.py

```diff
@@ -444,5 +444,53 @@ def test_float16_on_one_submodule(self):
             self.assertEqual(forward_inputs[c2].dtype, torch.float16)
 
 
+class TestFSDPModelCheckpointing(FSDPTest):
+    """Tests composable FSDP model checkpointing."""
+
+    @property
+    def world_size(self) -> int:
+        return 2
+
+    @skip_if_lt_x_gpu(2)
+    def test_state_dict_root_fully_shard(self):
+        """
+        Tests that the full state dict saved from a module with ``fully_shard``
+        applied to the global root matches that of an equivalent local module.
+        """
+        local_model = CompositeParamModel(device=torch.device("cuda"))
+        composable_module = copy.deepcopy(local_model)
+        fully_shard(composable_module, policy=ModuleWrapPolicy({UnitModule}))
```

Comment:
is it worth checking class based, I guess maybe not, due to transitive nature since we've verified class based can be equivalent to local?

Reply:
Yes, I think our goal is to match local, so we can test that directly. The transitivity holds as well.